Lib/test/test_statistics.py

```diff
@@ -2161,6 +2161,32 @@ def test_center_not_at_mean(self):
         self.assertEqual(self.func(data), 2.5)
         self.assertEqual(self.func(data, mu=0.5), 6.5)
 
+class TestSqrtHelper(unittest.TestCase):
+
+    def test_sqrt_frac(self):
+        # This helper function aspires to produce correctly rounded square roots of
+        # fractional inputs, more accurate than can be had with math.sqrt() alone.
+        # For a inputs spanning large ranges, we test that
+        # 1) the result is close to math.sqrt()
+        # 2) the result is at least as good as the two adjacent float values
+
+        randrange = random.randrange
+
+        for i in range(50_000):
```

Comment:
It may be worth adding some tests for the case where the numerator is significantly larger than the denominator (by a factor of around 10**40, say), to properly exercise the `q >= 0` branch of `_sqrt_frac`.

Reply:
FWIW, the tests currently do exercise that range and the code paths have 100% coverage.

Reply:
Yep, so they do. I misread the lines defining numerator and denominator.